awx/main/models/events.py

```diff
@@ -520,13 +520,21 @@ def _update_host_summary_from_stats(self, hostnames):
                 (summary['host_id'], summary['id'])
                 for summary in JobHostSummary.objects.filter(job_id=job.id).values('id', 'host_id')
             )
+            updated_hosts = set()
             for h in all_hosts:
                 # if the hostname *shows up* in the playbook_on_stats event
                 if h.name in hostnames:
                     h.last_job_id = job.id
+                    updated_hosts.add(h)
                 if h.id in host_mapping:
                     h.last_job_host_summary_id = host_mapping[h.id]
-            Host.objects.bulk_update(all_hosts, ['last_job_id', 'last_job_host_summary_id'])
+                    updated_hosts.add(h)
+
+            Host.objects.bulk_update(
+                list(updated_hosts),
+                ['last_job_id', 'last_job_host_summary_id'],
+                batch_size=100
```

Comment:
The other change is that I decided to split these updates into more reasonable chunks; I don't want to give up the performance benefits by calling host.save() per-Host, but it's also true that the list of hosts could be really large.

Reply:
I get the point of experimenting with `host.save()`, but for the purpose of finalizing this for merge, it would be a hard sell to use that as opposed to `bulk_update` with a batch size of, say, 10.

You're not in a transaction here, are you? If you are, I can see that undermining the goal of smaller batch sizes... but being in the callback receiver, I didn't think you were.

Reply:
Yea, in my testing, distinct `.save()` calls just aren't gonna work (too slow for large inventories).